xmtp_mls/src/client.rs

```diff
@@ -1489,14 +1509,19 @@ pub(crate) mod tests {
     >(
         client: &Client<ApiClient, Verifier>,
         installation_id: Id,
-    ) -> Vec<u8> {
-        let kps = client
+    ) -> Result<Vec<u8>, IdentityError> {
+        let kps_map = client
             .get_key_packages_for_installation_ids(vec![installation_id.as_ref().to_vec()])
             .await
-            .unwrap();
-        let kp = kps.first().unwrap();
+            .map_err(|_| IdentityError::NewIdentity("fetched malformed keypackage".to_string()))?;
+
+        let kp_result = kps_map
+            .get(installation_id.as_ref())
+            .ok_or(IdentityError::NewIdentity(
+                "fetched malformed keypackage".to_string(),
```

Comment:
Ideally we would persist this failure somewhere on the group so that someone can attempt to add the failed installation later. Otherwise we will update the sequence_id for the member in the `GroupMembership` and we'll never try to add this installation again.